src/drivers/distance_sensor/hxSRx0x/hxSRx0x_main.cpp

```diff
@@ -0,0 +1,210 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file hxSRx0x_main.cpp
+ * @author David Sidrane <david.sidrane@nscdg.com>
+ *
+ * Interface for the HY-SRF05 / HC-SR05 and HC-SR04.
+ * Precise Ultrasonic Range Sensor Module
+ */
+
+#include <cstdlib>
+#include <fcntl.h>
+#include <string.h>
+#include <stdio.h>
+#include <systemlib/err.h>
+
+#include <board_config.h>
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/module.h>
+
+#include "HxSRx0x.h"
+
+#if defined(HAVE_ULTRASOUND)
+
+/**
+ * @brief Local functions in support of the shell command.
+ */
+namespace hxSRx0x
+{
+
+HxSRx0x *instance = nullptr;
```

Comment:
You can use ModuleBase for most of this boilerplate now. Look at `drivers/adc/board_adc` for an okay example.

Basically instead of having `hxSRx0x_main.cpp` you just need to inherit `ModuleBase<HxSRx0x>`, then copy task_spawn, custom_command, print_usage.

https://github.com/PX4/PX4-Autopilot/blob/e6ad321ab2df10b8585d4ae4140b5ff5765966a9/src/drivers/adc/board_adc/ADC.hpp#L63-L77

https://github.com/PX4/PX4-Autopilot/blob/e6ad321ab2df10b8585d4ae4140b5ff5765966a9/src/drivers/adc/board_adc/ADC.cpp#L275-L337